bundler/lib/bundler/settings.rb

```diff
@@ -140,17 +146,22 @@ def set_global(key, value)
     end
 
     def all
-      keys = @temporary.keys | @global_config.keys | @local_config.keys | @env_config.keys
+      keys = @temporary.keys.union(@global_config.keys, @local_config.keys, @env_config.keys)
 
-      keys.map do |key|
-        key.sub(/^BUNDLE_/, "").gsub(/___/, "-").gsub(/__/, ".").downcase
-      end.sort
+      keys.map! do |key|
+        key = key.delete_prefix("BUNDLE_")
+        key.gsub!("___", "-")
+        key.gsub!("__", ".")
+        key.downcase!
+        key
+      end.sort!
+      keys
     end
 
     def local_overrides
       repos = {}
       all.each do |k|
-        repos[$'] = self[k] if k =~ /^local\./
+        repos[$'] = self[k] if k.start_with?("local.")
```

Comment:
Does the `$'` global get set when you didn't match a regexp?

I'm not sure what the simplest replacement is. Maybe...

```ruby
repos[k.delete_prefix("local.")] = self[k] if k.start_with?("local.")
```

Reply:
no it does not, good catch!

Reply:
ah-ha! that might be causing the test failures I'm seeing (I'm still bisecting to track down which introduced them)

Reply:
```suggestion
        repos[k.delete_prefix("local.")] = self[k] if k.start_with?("local.")
```

Reply:
I committed this change so we can see if the tests pass.

Reply:
Looks like they're passing now. yay!

Reply:
Awesome, thanks!